### DISCUSSION TITLE
75%-65% algo

### DISCUSSION THREAD

**Discussion by alex99r** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#166] [Ideas]
the battery optimal is 75-65. the algo should be able to stay at 75% (or apples 80%) then charge to 100% or stay at 75%. After some time (for example 2 hours) the battery should discharge to 65% then again spend some time there and then back to 75%. (this describes an ideal case of battery travelling to 75%-65% percentage maintaining the ultimate maintenance for battery longevity after the best charging combo. (other combos from best to worst include by ranking 75-45, 75-25, 85-25, 100-40, and many others.

**Comment by alex99r** [comment]
We can call this the charging algo since its the idea behind desktop mode of a connected a macbook pro with no support to macOS Battery Health Management. 

For this algo you need time left at idle charging state (i.e. 80% set) for 2 hours while charging. This time is also useful for another algo that could replicate Apple's implementation that is now missing from the unsupported models. Any strategy is possible by the user.

Congrats to the discharging feature hope you can bring it to Intel too soon. Cheers!!

**Reply by Allistah** [reply]
So in the context of this application, what is the best % to set the battery to then?

**Reply by alex99r** [reply]
not more than 80% and no less than 25%. But every once a month needs at least one 100% and one 0% full charge/discharge. Hence the best is to follow the first rule and go outside the limits every now and then based and how youre going to get the most out of it.

**Reply by alex99r** [reply]
75-65>75-45>75-35>80-40> any other combination possible

**Reply by danfoerster** [reply]
Alex! Please check my recent post on the discussion page. I think it could interest you because it would allow the battery to stay between 75 and 65% for a long time without cycling in-between at all, or too often for that matter. Cycling in between those two value will still wear and tear the battery as there are still small cycles created which add up to full cycles eventually as well. Of course it isn't as bad as always cycling in between 100 to completely empty, but still. In my opinion the best case scenario is if the battery holds a charge (anything in between 80 and 20%) but is neither drained nor charged by the user. Meaning it basically is just left alone to slowly self-discharge over a considerably long time. I don't think its necessary to have the battery percentage 'travel' around all that much, down to 65% and then again to 75 and so on and so forth, as you mentioned. I don't get the point of that, so if you could elaborate your thought pattern behind this please :)

**Reply by Allistah** [reply]
For what its worth, I'm using v2.1 alpha.  When I set mine to use the option "Use Classic SMC Key (Intel)" and have it set for 75%, the MacBook Pro stops charging and reports 78% with the plug icon.  It also states that the power source is the power adapter, and that the battery is not charging.  It has been rock solid at this number without any small range of cycling through discharging/charging.  After I set it to 75% though, I made sure that I rebooted and that it retained that setting and has been great since.

I got all sorts of mini-cycles when I didn't use that classic method which I didn't think was good.  It was constantly going up 2-3 %, down 2-3 %, all day long.

**Reply by danfoerster** [reply]
I am on the same version. When you had the classic SMC key mode unchecked, did you uncheck 'allow discharge'?
What Macbook do you have? If yours already has the build-in optimized battery charging and it is enabled I could see it conflicting with AlDente. Because one might want to bring it up to 80% and after that the other drains it back down to 78%.

**Reply by Allistah** [reply]
I had the option "Allow Discharge" turned on before I turned on the Classic SMC mode.  Maybe that was my mistake.

I'm on a 2015 15" MacBook Pro.

**Reply by danfoerster** [reply]
Classic SMC mode should eliminate the other option, as far as I know. Mine is a 2013 MacBook Pro, so almost identical. Do you use the original 85W power adapter?

**Reply by Allistah** [reply]
Yeah, I'm using an original Apple PSU.  It reads "85w Mag Safe 2 Power Adapter" on it.  The LED on the end of the cord always stays orange - it never goes green - even when it has stopped charging.

Are you using this without the Classis SMC Mode?  Do you have that "Allow Discharge" checked?  What is the behavior on your laptop for the LED?  Do you get any cycling of the state of charge at all?  Does it bounce up and down 1-3% from the set point?

**Reply by danfoerster** [reply]
Okay. I have the same adapter (I just wanted to check in case you were using a weaker 60W charger or something) and mine never goes green either. If it stays orange even when not charging, that's normal.
I have Classic SMC mode enabled. Therefore I cannot check/uncheck 'Allow Discharge' and it shouldn't have any effect anymore even if it was enabled before. My computer pretty much stays on the same percentage and does neither bounce up nor down.
Please read through my other post in the discussions called 'Top tip for users with stationary macbook'. There I have explained everything how I have mine set up at the moment.

**Reply by Allistah** [reply]
I turned SMC mode off, rebooted and will try it without the Allow Discharge turned on.  Will report back.  Having a hard time seeing where it will actually stop charging now.

**Reply by Allistah** [reply]
I couldn't get it to work at all without using the Legacy SMC mode.  It just acted like nothing was installed.  I uninstalled everything, rebooted, set it to 100% before deleting the app, reset the SMC, all of it and it only works with using that SMC mode.  Whatever, it works this way so I guess I'll just leave it alone.

**Reply by davidwernhart** [reply]
@jpirnie  which MacBook do you have/how old is it? maybe the new key simply does not exist on your machine.
best regards,
David

**Reply by Allistah** [reply]
It's a 2015 15" MacBook Pro.  The battery was just replaced by Apple about a week ago so it only has 4 cycles on it now.  It is running Big Sur 11.2.2.

**Reply by alex99r** [reply]
Assuming your laptop battery is now working as a ups battery then we can use the ups battery management. 

We know manufacturers suggest charging to 50% for long-term storage and that we should not charge past 80% generally while dropping below 50% is a reason to plug for a little bit

+ info suggests li-ion batteries need a full cycle of discharge a minimum every 90 days

**Reply by Allistah** [reply]
So I should really aim to set this battery at 50% then if I plan on using it all the time as a desktop.  I read that correctly, right?

**Reply by alex99r** [reply]
I think so

Doing a research and will report back

Interested in finding out other tests pointing to this since COVID became a thing

**Reply by alex99r** [reply]
> Alex! Please check my recent post on the discussion page. I think it could interest you because it would allow the battery to stay between 75 and 65% for a long time without cycling in-between at all, or too often for that matter. Cycling in between those two value will still wear and tear the battery as there are still small cycles created which add up to full cycles eventually as well. Of course it isn't as bad as always cycling in between 100 to completely empty, but still. In my opinion the best case scenario is if the battery holds a charge (anything in between 80 and 20%) but is neither drained nor charged by the user. Meaning it basically is just left alone to slowly self-discharge over a considerably long time. I don't think its necessary to have the battery percentage 'travel' around all that much, down to 65% and then again to 75 and so on and so forth, as you mentioned. I don't get the point of that, so if you could elaborate your thought pattern behind this please :) 
> 
> 
> 
> 

You're probably right !!

**Reply by alex99r** [reply]
No point in traveling in cycles. Keep an SoC between 40-80 based on literature. Voltage increases with SoC. I am trying find more info on the best stable SoC and what matters. Don't forget to cycle every 30-90 days at least once.

LOL: So if you are having issues with throttling maybe set SoC towards 40-50 or if you're over clocking and need more juice increase the SOC.

**Reply by alex99r** [reply]
Charging the battery from 80 to 100 is based on the premise of going out and using it away from a plug, just like a phone. 

Not having control over that seems totally archaic to run to the wall to charge your juice and then unplug it to consume it right sat in the same position.

Need control over desktop mode and going out in 30 Mins mode. Just like Apple HBM system but more conscious about Laptop at home. Today we found out a simple 50% fixed SoC was all the fuck we wanted from the start with an ability to charge fully cause I'm going out and I'm taking it with me.

**Reply by Allistah** [reply]
> @jpirnie which MacBook do you have/how old is it? maybe the new key simply does not exist on your machine.
> best regards,
> David

Looks like the latest version v2.1.2 has fixed the issues I was seeing.  I made sure to turn off allow discharge and it seems to be staying where it should.  I'll let it run for a day and see what it looks like.

Maybe another option would be to let that option to discharge to stay on until it reaches the destination charge %, then automatically turn it off.  Otherwise you get that weird cycling thing going on.
